station_performance_pipeline/extract.py

```diff
@@ -0,0 +1,89 @@
+from os import environ as ENV
+from datetime import datetime
+
+import requests
+from requests.auth import HTTPBasicAuth
+from dotenv import load_dotenv
+
+
+def train_station_arrival_data(station_crs: str) -> list[dict]:
+    '''This function accepts the crs code of a station and returns a list of dictionaries.
+    Each dictionary represents a train that arrived at the station on this day.
+    The dictionary has the keys related to the expected and actual times of arrival and departure.'''
+
+    now = datetime.now()
+    response = requests.get(
+        F"https://api.rtt.io/api/v1/json/search/{station_crs}/{now.strftime('%Y')}/{now.strftime('%m')}/{now.strftime('%d')}", auth=HTTPBasicAuth(ENV["REALTIME_API_USER"], ENV["REALTIME_API_PASS"]))
+
+    todays_station_data = response.json()
+
+    # Data on all of the instances of services running stopping at teh station
+    services_data = todays_station_data["services"]
```

Comment:
Does inputting a massive amount of data into a function make it slower? Although, I agree that it probably be quicker than a second api call. Good idea

Reply:
@isaacschaessenscoleman From my understanding the input here is actually a reference to the dictionary, rather than the dictionary itself. Its pass by reference vs pass by value which is a lot more performant.